src/sage/sets/disjoint_set.pyx

```diff
@@ -379,28 +376,29 @@ cdef class DisjointSet_of_integers(DisjointSet_class):
             sage: d = DisjointSet(5)
             sage: d.__getstate__()
             [0, 1, 2, 3, 4]
-            sage: d.union(2,3)
+            sage: d.union(2, 3)
             sage: d.__getstate__()
             [0, 1, 2, 2, 4]
-            sage: d.union(3,0)
+            sage: d.union(3, 0)
             sage: d.__getstate__()
             [2, 1, 2, 2, 4]
 
         Other parents are obtained when the operations are done is a
         distinct order::
 
             sage: d = DisjointSet(5)
-            sage: d.union(0,3)
+            sage: d.union(0, 3)
             sage: d.__getstate__()
             [0, 1, 2, 0, 4]
-            sage: d.union(2,0)
+            sage: d.union(2, 0)
             sage: d.__getstate__()
             [0, 1, 0, 0, 4]
         """
-        l = []
+        cdef Py_ssize_t card = self.cardinality()
+        cdef list l = [None] * card
         cdef int i
-        for i in range(self.cardinality()):
-            l.append(self._nodes.parent[i])
+        for i in range(card):
+            l[i] = self._nodes.parent[i]
```

Comment:
wouldn't `l = [self._nodes.parent[i] for i in range(self.cardinality())]` be better?

Hm, probably not: https://github.com/cython/cython/issues/2844

Reply:
Yes, I doubt it would help.